pandas/tests/apply/test_str.py

```diff
@@ -30,10 +30,10 @@
 )
 @pytest.mark.parametrize("how", ["agg", "apply"])
 def test_apply_with_string_funcs(request, float_frame, func, args, kwds, how):
-    if len(args) > 1 and how == "agg":
+    if len(args) > 1:
         request.node.add_marker(
             pytest.mark.xfail(
-                raises=TypeError,
+                raises=TypeError if how == "agg" else FutureWarning,
```

Comment:
I think this only fails because we configure pytest to raise internal pandas warnings as errors. Could you find a way to test the warning instead (or refactor the test entirely since it's tested above)?

Reply:
I've made a new version.